### ISSUE TITLE
Handle snake case method at relationship model

### ISSUE THREAD

**Issue by mishbah** [lukeraymonddowning/poser#77]
I have model like this

```php
<?php

namespace App;

class ChildModel extend Model
{
    public function parent_model()
    {
        return $this->belongsTo('App\ParentModel');
    }
}

class ParentModel extend Model
{
}
```

And I create `ChildModelFactory` and `ParentModelFactory`. When I write the `ChildModelFactory` at test section:

```php
$test = ChildModelFactory::new()->forParentModel()->create();
dd($test);
```
and when run `phpunit`, I show the error

```
BadMethodCallException: Call to undefined method App\ChildModel::parentModel()
```
The error gone, when I replace method name from `parent_model` to `parentModel`. My question is...
Is there a way to resolve the problem without changing the method name?

**Comment by lukeraymonddowning** [comment]
@mishbah thanks for this. I've just released `4.0.1`, which solves your issue. It will now look for your `snake_case` relationship when resolving.

**Comment by lukeraymonddowning** [comment]
@all-contributors add @mishbah for ideas
